content/blog/production-ready-object-detection-model-training-workflow-with-hpe-machine-learning-development-environment.md

```diff
@@ -0,0 +1,1107 @@
+
+Here is what a configuration file looks like for a distributed training experiment.
+
+Below is what the `determined_files/const-distributed.yaml` contents look like. `slots_per_trial: 8` defines that we will use 8 GPUs for this experiment.
+
+Edit the change the workspace and project settings in the `determined_files/const-distributed.yaml` file
+
+```yaml
+name: resnet_fpn_frcnn_xview_dist_warmup
+workspace: <WORKSPACE_NAME>
+project: <PROJECT_NAME>
+profiling:
+ enabled: true
+ begin_on_batch: 0
+ end_after_batch: null
+hyperparameters:
+    lr: 0.01
+    momentum: 0.9
+    global_batch_size: 128
+    # global_batch_size: 16
+    weight_decay: 1.0e-4
+    gamma: 0.1
+    warmup: linear
+    warmup_iters: 200
+    warmup_ratio: 0.001
+
+    step1: 18032 # 14 epochs: 14*1288 == 18,032
+    step2: 19320 # 15 epochs: 15*1288 == 19,320
+    model: fasterrcnn_resnet50_fpn
+    # Dataset
+    dataset_file: coco
+    backend: aws # specifiy the backend you want to use.  one of: gcs, aws, fake, local
+    data_dir: determined-ai-coco-dataset # bucket name if using gcs or aws, otherwise directory to dataset
+    masks: false
+    num_workers: 4
+    device: cuda
+environment:
+    image: determinedai/environments:cuda-11.3-pytorch-1.10-tf-2.8-gpu-mpi-0.19.10
+    environment_variables:                                                                          
+        - NCCL_DEBUG=INFO                                                                           
+        # You may need to modify this to match your network configuration.                          
+        - NCCL_SOCKET_IFNAME=ens,eth,ib
+bind_mounts:
+    - host_path: /tmp
+      container_path: /data
+      read_only: false
+scheduling_unit: 400
+min_validation_period:
+    batches: 1288 # For training
+
+searcher:
+  name: single
+  metric: mAP
+  smaller_is_better: true
+  max_length:
+    batches: 38640 # 30*1288 == 6440# Real Training
+records_per_epoch: 1288
+resources:
+    slots_per_trial: 8
+    shm_size: 2000000000
+max_restarts: 0
+
+entrypoint: python3 -m determined.launch.torch_distributed --trial model_def:ObjectDetectionTrial
+```
+
+```python
+# Run the below cell to kick off an experiment
+!det e create determined_files/const-distributed.yaml determined_files/
+```
+
+```
+Preparing files to send to master... 237.5KB and 36 files
+Created experiment 77
+```
+
+## Launching a Distributed Hyperparameter Search Experiment
+
+To implement an automatic hyperparameter tuning experiment, we need to define the hyperparameter space, e.g., by listing the decisions that may impact model performance. We can specify a range of possible values in the experiment configuration for each hyperparameter in the search space.
+
+View the `x.yaml` file that defines a hyperparameter search where we find the model architecture that achieves the best performance on the dataset.
```

Comment:
You can either change "we" to "you" or do the following (which would actually be better):

"View the 'x.yaml' file that defines a hyperparameter search where the model architecture that achieves the best performance on the dataset is found." (yes, it's passive tense, but it gets rid of the Royal We.)